### DISCUSSION TITLE
1.2 beta 2, Cubase Async issue and solution

### DISCUSSION THREAD

**Discussion by igbro** [apohl79/audiogridder#524] [General]
First of all, congrats to Andreas, I made the big jump from beta 4 (the old working guy for me) to this last 1.2 beta 2.

I'm running Cubase (last 11.0.20 Pro win10) at 1 (yes one) block with 16 AA big plugins on the same AG server instance (dual xeon win7)!!

Very fluent with Vnc (remote desktop gave me some strange things). 
Buffer is set to 1024 but for my RME, it only means 23ms latency@44.1. So AG, in theory, only adds another 23ms. (hosted plugins latency,must be added)
Asio Guard is set to high. individually, AG plugins have ASIO Guard enabled! ( who says, you have to disable VEP!!!)
Both machines are 8-10 years old!
Direct gigabit wiring.
Got only 10% CPU taxed on the server for 16 AA console strips.(2x 6(12) cores@3ghz)
Sandboxing and parallel loading are set to on in AG server.

Stunning success!

And yes, like I read in other threads, sometimes, Cubase plays some tracks not in sync.
I've seen that this issue never comes with the first project you load after Cubase launch, but starting with the second one and so on.

In this case, I found you have to force latency redetection of Agridder/Audiogridder plugins.

In Studio => VST plugins manager panel, just enable/disable or disable/enable Asio Guard for the Audiogridder plugins and everything comes back in sync.

I'm sure other Daw offer some, (maybe not obvious) way to force plugin latency redetection.

Technically, for VST's, I don't think it's a dynamic thing and rather only reported by the plugin when a query is done by the host.

It must be a mess for a "wrapper" like AG to report latency because it depends on which plugins is hosted and this, each time you load a song and even between each instances of the same song!
The DAW cant' firmly associate a VST id with a fixed latency.
Reporting the highest latency of all instances of the song, is certainly the way to go, but Cubase is messing around with this between songs.

Hope it helps. Cheers.

Igor.

**Comment by apohl79** [comment]
that's great! thx for the info! i will try to reproduce this, but it sounds like ag has to enforce the sync of the sample latency to the daw every now and than. usually that happens on every audio chunk coming back from the server. but the daw will only get an update on a changed value. maybe there is some scenarios, where the plugin is under the wrong assumption. anyways, worth a try i guess. an alternative could be a "resync" command, that does what you are enforcing via the settings workaround.

cheers

**Comment by igbro** [comment]
Yes Andreas, and I confirm that at the very moment (working on a mix). 

When I add another plugin to the chain of an instance of an AG plugin, this one and only track gets out of sync.

Once again, forcing detection with the ASIO Guard trick did resync everything.

The new latency has thus not been reported properly to the DAW.

**Comment by igbro** [comment]
Hallo Andreas,

I dug a bit further on this issue.

This has to do with Asio Guard which apparently prevents from recalculating latency when set to on.

When I permanently disengage Asio Guard (the double buffering system) on the AG plugin, latency changes are properly reported to Cubase. (when adding a new plugin in the AG chain for instance).

I think you already know Cubase, but you can individually set double buffering on/off for a specific plugin.
That's what I'm talking about, not the overall Asio Guard setting.

This setting cranks up the "max" Asio meter level in Cubase but leaves the "average" one as it was before. 

It definitely stresses Asio, but remains workable (with higher blocks settings) and re enabling Asio Guard is still possible after all AG plugins have been inserted.

Having Asio Guard on for the AG plugin is why I'm able to run @ 1 block with 16 Acustica Console strips plugins.

With VEP, some forums guys say it is advisable to disable Asio Guard for the Vienna plugin because of the LAN thing, but I never had such issues leaving it on with my V.7 and the exact same gear.

In this regard, this may be a known Cubase issue.

Cheers.

**Reply by apohl79** [reply]
thx a lot for this! in theory AG could use some headroom latency buffer that inserted plugins can "eat" up, until a recalculation and a daw update would be needed. that could help with this issue.

**Reply by AtmanActive** [reply]
As long as it is end-user configurable, this setting could be very useful. 
I guess sensible options would be:
- Auto - as it is now, where AG reports PDC back dynamically
- Static (1 - 20) - where AG would report latency of X buffers to DAW and then internally delay the whole chain up to X buffers. Hence, if end user expects plugins in the range of, let's say, 2 buffers, and the DAW is set to 512, then, AG would report 1024 samples PDC to DAW and start delaying audio for 1024. As soon as AG starts hosting a plugin, of, let's say 512 samples PDC it would reduce it's own delay line to 512 while DAW still holds 1024 as true. But, if a plugin iz added to the chain that has, for example, 2048 reported PDC, then, AG would still keep 1024 reported to DAW, keep it's internal delay line at 0 and show +1024spls in red letters in it's UI.

In other words, if needed, let the end-user manage PDC predictably.

Just thinking out loud here.

**Reply by apohl79** [reply]
yep, exactly!

**Comment by igbro** [comment]
Hallo Andreas,

GOOD NEWS!
The latency report war against your AG and the Steinberg AG (Asio Guard) is over when using the VST3 AGridder plugin instead of the VST2 one.

I've tried everything with VST2 AG version.
Disabling the individual Asio Guard setting finally did not correctly report new latencies to Cubase either.
Completely disabling Asio Guard in Cubase helped, but this is a no go option for big projects.
This is broken from the AG 1.1 beta 4 I used a lot without this issue.

The enforcing latency trick indeed re-syncs everything but, at the "building" stage, it has to be done at each AG/Hosted plugin insertion.

So I tried to insert VST3 AG plugin instead of VST2, saving each chain preset from VST2s and reloading it in each new VST3s.

And Voilà...

You can insert/delete AG plugin, change hosted ones and every thing remains in sync with Asio Guard enabled everywhere.

I don't know why but I was exclusively using the VST2... until now.

SAD NEWS!
Because paradise is not that easy to reach, VST3 gives me a strange hosted plugin GUI size detection issue.

Please, watch my video showing VST2 vs VST3 opening the hosted plugin GUI.

VST3 needs 2 attempts.
This a blank Cubase project (1.2 beta 3. DAW W10 / Server W7)

https://user-images.githubusercontent.com/65025818/117688270-ea6c5880-b1b8-11eb-9c0b-de90ba7a5dd8.mp4

Cheers.

Igor.

**Reply by apohl79** [reply]
Thx, created issue https://github.com/apohl79/audiogridder/issues/546

**Comment by igbro** [comment]
Just want to add... 
On server side, plugins are all VST2's...
